cluster-autoscaler/clusterstate/testutils.go

```diff
@@ -20,18 +20,47 @@ import (
 	"time"
 
 	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	k8scloudprovider "k8s.io/cloud-provider"
 )
 
-// NewMockMaxNodeProvisionTimeProvider returns static maxNodeProvisionTimeProvider which returns constant MaxNodeProvisionTime for every NodeGroup.
-func NewMockMaxNodeProvisionTimeProvider(maxNodeProvisionTime time.Duration) *staticMockMaxNodeProvisionTimeProvider {
-	return &staticMockMaxNodeProvisionTimeProvider{maxNodeProvisionTime}
+// NewMockMaxNodeProvisionTimeProcessor returns static NodeGroupConfigProcessor which returns constant MaxNodeProvisionTime for every NodeGroup.
```

Comment:
Wouldn't this sort of functionality be better placed in the same package as the processor itself? It's not inherently connected to CSR.

I'm also not convinced we need it at all in most places - any test that is not testing overriding features would get the same behavior with the default implementations and setting the value in NodeGroupDefaults, right? But that's a smaller issue - it's repeated in a lot of places, so maybe the syntactic sugar is worth it.

Reply:
I've removed the static mock entirely, that was a good suggestion.